spec/functional/resource/env_spec.rb

```diff
@@ -22,9 +22,27 @@
   context "when running on Windows" do
     let(:chef_env_test_lower_case) { "chefenvtest" }
     let(:chef_env_test_mixed_case) { "chefENVtest" }
+    let(:chef_env_with_delim) { "chef_env_with_delim" }
+    let(:chef_env_delim) { ";" }
+    let(:chef_env_test_delim) { "#{value1};#{value2}" }
     let(:env_dne_key) { "env_dne_key" }
     let(:env_value1) { "value1" }
     let(:env_value2) { "value2" }
+    let(:delim_value) { "#{env_value1};#{env_value2}" }
+    let(:env_user) { ENV["USERNAME"].upcase }
+    let(:default_env_user) { "<SYSTEM>" }
+
+    let(:env_obj) do
+      wmi = WmiLite::Wmi.new
+      environment_variables = wmi.query("select * from Win32_Environment where name = '#{test_resource.key_name}'")
+      if environment_variables && environment_variables.length > 0
+        environment_variables.each do |env|
+          @env_obj = env.wmi_ole_object
+          return @env_obj if @env_obj.username.split('\\').last.casecmp(test_resource.user) == 0
+        end
+      end
+      @env_obj = nil
```

Comment:
Please return a value from this method instead of setting the instance variable.